tests/functional/src/main/java/org/mule/functional/junit4/runners/ArtifactClassloaderTestRunner.java

```diff
@@ -0,0 +1,125 @@
+        if (runnerDelegateTo != null)
+        {
+            runnerClass = runnerDelegateTo.value();
+        }
+        runners.add(runnerClass.cast(runnerClass.getConstructor(Class.class).newInstance(getTestClass().getJavaClass())));
+    }
+
+    @Override
+    protected List<Runner> getChildren() {
+        return runners;
+    }
+
+    @Override
+    public Description getDescription() {
+        // Just to avoid having duplicated name in IDEA/Eclipse JUnit panel view due to this is a Suite, we always have only one delegate/child runner.
+        return runners.get(0).getDescription();
+    }
+
+
+    protected void runChild(final Runner runner, final RunNotifier notifier) {
+        withContextClassLoader(getTestClass().getJavaClass().getClassLoader(), () -> runner.run(notifier));
+    }
+
+    /**
+     * @param klass
+     * @return the {@link ClassLoader} that would be used to run the test. This way the test will be isolated and it will behave
+     * similar as an application running in a Mule standalone container.
+     * @throws Throwable
+     */
+    private static Class<?> createTestClassUsingClassLoader(Class<?> klass) throws Throwable
+    {
+        // Initializes utility classes
+        ClassPathURLsProvider classPathURLsProvider = new DefaultClassPathURLsProvider();
+        MavenDependenciesResolver mavenDependenciesResolver = new DependencyGraphMavenDependenciesResolver();
+        MavenMultiModuleArtifactMapping mavenMultiModuleArtifactMapping = new MuleMavenMultiModuleArtifactMapping();
+        ClassLoaderRunnerFactory classLoaderRunnerFactory = new MuleClassLoaderRunnerFactory();
+        ClassPathClassifier classPathClassifier = new MuleClassPathClassifier();
+
+        // Does the classification and creation of the isolated ClassLoader
+        ArtifactUrlClassification artifactUrlClassification = classPathClassifier.classify(new DefaultClassPathClassifierContext(klass, classPathURLsProvider.getURLs(),
+                                                                                           mavenDependenciesResolver.buildDependencies(), mavenMultiModuleArtifactMapping));
+        ClassLoaderTestRunner classLoaderTestRunner = classLoaderRunnerFactory.createClassLoader(klass, artifactUrlClassification);
+
+        Class<?> isolatedTestClass = classLoaderTestRunner.loadClassWithApplicationClassLoader(klass.getName());
+
+        injectPluginsClassLoaders(classLoaderTestRunner, isolatedTestClass);
+
+        return isolatedTestClass;
+    }
+
+    /**
+     * Invokes the method to inject the plugin/extension classloaders for registering the extensions to the {@link org.mule.runtime.core.api.MuleContext}.
+     * @param classLoaderTestRunner
+     * @param isolatedTestClass
+     * @throws Throwable
+     */
+    private static void injectPluginsClassLoaders(ClassLoaderTestRunner classLoaderTestRunner, Class<?> isolatedTestClass) throws Throwable
+    {
+        TestClass testClass = new TestClass(isolatedTestClass);
```

Comment:
Seems like a plain Object will work instead of wrapping the class in a TestClass